server/src/infra/repositories/machine-learning.repository.ts

```diff
@@ -1,21 +1,45 @@
 import { DetectFaceResult, IMachineLearningRepository, MachineLearningInput, MACHINE_LEARNING_URL } from '@app/domain';
 import { Injectable } from '@nestjs/common';
 import axios from 'axios';
+import FormData from 'form-data';
+import { createReadStream } from 'fs';
 
 const client = axios.create({ baseURL: MACHINE_LEARNING_URL });
 
 @Injectable()
 export class MachineLearningRepository implements IMachineLearningRepository {
+  private createFormData(input: MachineLearningInput): FormData {
+    const formData = new FormData();
+    const fileStream = createReadStream(input.imagePath);
+    formData.append('image', fileStream);
+    return formData;
+  }
```

Comment:
Rather than returning the FormData, if you change this function to also accept the HTTP request path then you can include the whole `client.post(...)` call in here as well for even more deduplication.

Reply:
but all three methods return different type of data, so we will have to return `Promise<any>` from common method and for example cast it to `Promise<string[]>` for classifyImage  

If you are good with that, I can make the changes

Reply:
Good point, I missed that. That could be solved by using generics but this is fine for now tbh.

Reply:
```ts
  private makeAPICall(input: MachineLearningInput, endpoint:string): Promise<any> {

    const formData = new FormData();
    const fileStream = createReadStream(input.imagePath);
    formData.append('image', fileStream);
    return client
      .post(endpoint, formData, {
        headers: formData.getHeaders(),
      })
      .then((res) => res.data);
  }

  classifyImage(input: MachineLearningInput): Promise<string[]> {
    return this.makeAPICall(input, '/image-classifier/tag-image');
  }

  detectFaces(input: MachineLearningInput): Promise<DetectFaceResult[]> {
    return this.makeAPICall(input, '/facial-recognition/detect-faces');
  }

  encodeImage(input: MachineLearningInput): Promise<number[]> {
    return this.makeAPICall(input, '/sentence-transformer/encode-image');
  }
```

looks clean though!